transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/TransactionOutbox.java

```diff
@@ -202,52 +202,54 @@ private void expireIdempotencyProtection(Instant now) {
   }
 
   /**
-   * Marks a blacklisted entry back to not blacklisted and resets the attempt count. Requires an
-   * active transaction and a transaction manager that supports thread local context.
+   * Unblocks a blocked entry and resets the attempt count so that it will be
+   * retried again. Requires an active transaction and a transaction manager that supports thread
+   * local context.
    *
    * @param entryId The entry id.
-   * @return True if the whitelisting request was successful. May return false if another thread
-   *     whitelisted the entry first.
+   * @return True if the request to unblock the entry was successful. May return
+   *     false if another thread unblocked the entry first.
    */
-  public boolean whitelist(String entryId) {
+  public boolean unblock(String entryId) {
     if (!(transactionManager instanceof ThreadLocalContextTransactionManager)) {
       throw new UnsupportedOperationException(
           "This method requires a ThreadLocalContextTransactionManager");
     }
-    log.info("Whitelisting entry {}", entryId);
+    log.info("Marking entry {} for retry.", entryId);
     try {
       return ((ThreadLocalContextTransactionManager) transactionManager)
-          .requireTransactionReturns(tx -> persistor.whitelist(tx, entryId));
+          .requireTransactionReturns(tx -> persistor.unblock(tx, entryId));
     } catch (Exception e) {
       throw (RuntimeException) Utils.uncheckAndThrow(e);
     }
   }
 
   /**
-   * Marks a blacklisted entry back to not blacklisted and resets the attempt count. Requires an
-   * active transaction and a transaction manager that supports supplied context.
+   * Clears a failed entry of its failed state and resets the attempt count so that it will be
+   * retried again. Requires an active transaction and a transaction manager that supports supplied
+   * context.
    *
    * @param entryId The entry id.
    * @param transactionContext The transaction context ({@link TransactionManager} implementation
    *     specific).
-   * @return True if the whitelisting request was successful. May return false if another thread
-   *     whitelisted the entry first.
+   * @return True if the request to clear the failed state of the entry was successful. May return
```

Comment:
True if the unblock request was successful. May return false if another thread unblocked the entry first.